Learning_test_automation/Projekt#1/TeachersGradebook/TeachersRoom/GradeFile.cs

```diff
@@ -0,0 +1,78 @@
+﻿using SchoolData;
+
+namespace Infrastructure
+{
+    public class GradeFile
+    {
+        FileWizard fileWizard = new FileWizard();
+
+        public void GradeFileCreation(string createText)
+        {
+            string uniqName = "Grade";
+            fileWizard.SaveDataToFile(uniqName, createText);
+        }
+
+        public void AddGradeParametersFromFileToList()
+        {
+            string fileName = @"C:\Users\lemk\Desktop\TeachersGradebook\Grade.txt";
+
+            if (File.Exists(fileName))
+            {
+                using (StreamReader sr = new StreamReader(fileName))
+                {
+                    if (sr.Peek() >= 0)
+                    {
+                        string fileContent = sr.ReadToEnd();
+
+                        string[] grades = fileContent.Split('\n');
+
+                        SchoolSubjects schoolSubject = default;
+                        double gradeValue    = default;
+                        int studentId     = default;
+                        int gradeID = default;
+
+                        foreach (var lineWithGradeDate in grades)
+                        {
+                            string[] fields = lineWithGradeDate.Split(';');
+
+                            foreach (string field in fields)
+                            {
+                                if (field.Trim().StartsWith("School subject:"))
+                                {
+                                    SchoolSubjects.TryParse(field.Trim().Substring("School subject:".Length).Trim(), out schoolSubject);
+                                }
```

Comment:
Ten if i pozostale trzy, wygladaja bardzo podobnie, wyglada na bardzo dobrego kandydata osobnej metody, ktora przyjmowalaby parametry i zwracala wartosc :)

Reply:
Ale zauważyłeś że prawie każdy TryParse ma inny typ który pobiera i który zwraca. Jesli można to ogarnąć to chętnie zobaczę